luxonis_train/nodes/heads/base_head.py

```diff
@@ -0,0 +1,77 @@
+from abc import abstractmethod
+from typing import Generic, Literal
+
+from luxonis_train.nodes.base_node import (
+    BaseNode,
+    ForwardInputT,
+    ForwardOutputT,
+)
+from luxonis_train.utils import deep_merge_dicts
+
+
+class BaseHead(
+    BaseNode[ForwardInputT, ForwardOutputT],
+    Generic[ForwardInputT, ForwardOutputT],
+):
+    """Base class for all heads in the model.
+
+    @type parser: str | None
+    @ivar parser: Parser to use for the head.
+    @type is_softmaxed: bool | None
+    @ivar is_softmaxed: Whether the head uses softmax or not.
+    """
+
+    parser: (
+        Literal[
+            "ClassificationParser",
+            "YOLO",
+            "YoloDetectionNetwork",
+            "SegmentationParser",
+        ]
+        | None
+    ) = None
+    is_softmaxed: bool | None = None
```

Comment:
IMO this should this be specifically added only to the heads where relevat (as we do for example with `iou_threshold` on detection head)